engine/src/main/java/com/datatorrent/stram/engine/OperatorContext.java

```diff
@@ -84,12 +87,13 @@ public void setIdleTimeout(long idleTimeout)
    * @param attributes the value of attributes
    * @param parentContext
    */
-  public OperatorContext(int id, AttributeMap attributes, Context parentContext)
+  public OperatorContext(int id, String operatorName, AttributeMap attributes, Context parentContext)
   {
     super(attributes, parentContext);
     this.lastProcessedWindowId = Stateless.WINDOW_ID;
     this.id = id;
     this.stateless = super.getValue(OperatorContext.STATELESS);
+    this.operatorName = Preconditions.checkNotNull(operatorName, "operator name");
```

Comment:
Check for OperatorDeployInfo not null is optional as it will be necessary to dereference OperatorDeployInfo.id and OperatorDeployInfo.name in the OperatorContext constructor, so NullPointerException will be raised anyway. Annotating OperatorDeployInfo parameter as `@NotNull` is good, but we don't follow this practice in other places.